### PR TITLE
Svg linear gradient

### PR DESCRIPTION
**Pull request by james-lawrence**
based on #1970

before:
screenshot-2021-09-15-153848

after:
screenshot-2021-09-15-153928

goal:
screenshot-2021-09-15-153711

### CODE REVIEW

**Comment by jneem** [comment]
Is this different from #1970? It seems to have both radial and linear?

**Comment by james-lawrence** [comment]
sorry it is both as its based off of #1970.

File: druid/src/widget/svg.rs [hunk 1]
```diff
@@ -354,32 +354,54 @@ impl SvgRenderer {
...
             usvg::Paint::Color(c) => {
                 // TODO I'm going to assume here that not retaining colors is OK.
                 let color = color_from_svg(*c, opacity);
-                Rc::new(ctx.solid_brush(color))
+                Rc::new(piet::PaintBrush::Color(color))
             }
-            usvg::Paint::Link(id) => self.defs.find(id).unwrap(),
+            usvg::Paint::Link(id) => match self.defs.find(id) {
+                None => {
+                    // generally this occurs due to unimplement SVG functionality.
```
Comment (jneem) [inline druid/src/widget/svg.rs:391]: unimplemented

**Review by jneem — APPROVED** [review]
Looks good, thanks!

**Pull request merged**